src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java

```diff
@@ -123,9 +124,11 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
 
         final String nodeName = buildData.getNodeName("").isEmpty() ? pipelineData.getNodeName("") : buildData.getNodeName("");
         buildSpan.setTag(CITags.NODE_NAME, nodeName);
-
-        final String nodeHostname = buildData.getHostname("").isEmpty() ? pipelineData.getHostname("") : buildData.getHostname("");
-        buildSpan.setTag(CITags._DD_HOSTNAME, nodeHostname);
+        // If the NodeName == master, we don't set _dd.hostname. It will be overridden by the Datadog Agent. (Traces are only available using Datadog Agent)
```

Comment:
That's not true when the plugin is configured in HTTP mode. In that case, no hostname will be reported.

Reply:
Yeah, Jenkins traces collection is only available if the plugin is configured in a Datadog Agent mode: ([link](https://github.com/jenkinsci/datadog-plugin/blob/master/src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java#L659-L675)), so we can assume that's correct, right?

Reply:
Oh right, completely forgot about that. Was there any plan to allow sending traces via HTTP? If yes we should add some additional logic here to fail if the plugin runs in http mode. That will help in the future